src/url.cpp

```diff
@@ -210,6 +210,13 @@ namespace Url
             host_.assign(url, position, index - position);
             position = index;
 
+            if (    host_.empty()
```

Comment:
Could you also drop this extra whitespace between the open paren and the rest of the conditions?